corehq/apps/translations/management/commands/translate_po_files.py

```diff
@@ -0,0 +1,679 @@
+        Keep the structure and formatting of the original text."""
+        return base_prompt
+
+    def input_format_prompt(self):
+        return f"Input format: {self.translation_format.format_input_description()}"
+
+    def output_format_prompt(self):
+        return f"Output format: {self.translation_format.format_output_description()}"
+
+    @abc.abstractmethod
+    def supported_models(self):
+        return []
+
+    def translate(self, input_data):
+        system_prompt = "\n".join([
+            self.base_prompt(),
+            self.input_format_prompt(),
+            self.output_format_prompt(),
+        ])
+        user_message = self.translation_format.format_input(input_data)
+
+        llm_output = self._call_llm(system_prompt, user_message)
+        return self.translation_format.parse_output(llm_output)
+
+    @abc.abstractmethod
+    def _call_llm(self, system_prompt, user_message):
+        """Call LLM using the client library"""
+        pass
+
+    @abc.abstractmethod
+    def _call_llm_http(self, system_prompt, user_message):
+        """Call LLM using direct HTTP requests without client libraries"""
+        pass
+
+
+class TranslationFormat(abc.ABC):
+    """
+    Abstract class for different translation formats.
+    The idea is to have a class for each format and have input prompt and output prompt for each format.
+    Defined in the subclasses. It also has methods to load input, format input, parse output, save output.
+    An example can be we can have a class for Simple text file, JSON file etc.
+    We have implemented a class for PO file translation.
+    """
+    @abc.abstractmethod
+    def load_input(self, input_source=None):
+        pass
+
+    @abc.abstractmethod
+    def format_input(self, input_data):
+        pass
+
+    @abc.abstractmethod
+    def parse_output(self, output_data):
+        pass
+
+    @abc.abstractmethod
+    def save_output(self, output_data, output_path):
+        pass
+
+    @abc.abstractmethod
+    def format_input_description(self):
+        pass
+
+    @abc.abstractmethod
+    def format_output_description(self):
+        pass
+
+
+class OpenaiTranslator(LLMTranslator):
+    @property
+    def supported_models(self):
+        return ["gpt-4o-mini", "gpt-4o", "gpt-4", "gpt-4.1"]
+
+    def __init__(self, api_key, model, lang, translation_format, backup_model=''):
+        super().__init__(api_key, model, lang, translation_format)
+        try:
+            import openai
+            self.openai = openai
+            self.client = openai.OpenAI(api_key=api_key)
+        except ImportError:
+            self.openai = None
+            self.client = None
+            print("OpenAI Python package not found, will use HTTP requests instead.")
+
+        self.api_base = "https://api.openai.com/v1"
+
+    def _call_llm(self, system_prompt, user_message):
+
+        if self.client is None:
+            return self._call_llm_http(system_prompt, user_message)
+        else:
```

Comment:
Nitpick: since the `if` block ends with a return, you could remove the else and drop a level of indentation.

Reply:
https://github.com/dimagi/commcare-hq/pull/36335/commits/e84c27bd405acef216468e05003040192591c124